torch/csrc/autograd/python_function.cpp

```diff
@@ -94,9 +94,13 @@ auto PyFunction::legacy_apply(const variable_list& inputs) -> variable_list {
   // leads to unexpected error messages ("no nodes require computing gradients"),
   // but I don't have a better idea. These functions would raise an error
   // in backward anyway.
-  return wrap_outputs(inputs, std::move(tensor_results), [this](FunctionFlags &&f) {
-    return std::make_shared<Error>(name() + " is not differentiable twice", std::move(f));
-  });
+  return wrap_outputs(
+      inputs,
+      std::move(tensor_results),
+      [this](function_list&& next_functions) {
+        return std::make_shared<Error>(
+            name() + " is not differentiable twice", std::move(next_functions));
```

Comment:
Now I realized that this code can lead to a segfault I think, because the function we return doesn't own `this`, which is needed by `name()`. Can you just capture it by value?

Reply:
This should be fine since `wrap_outputs` is evaluated right here and calls the lambda inside, all before returning. So it's still cheaper to take `this`

Reply:
Ah, good point!